spec/models/replication_transfer_spec.rb

```diff
@@ -140,12 +141,32 @@
     end
     it "cannot cancel a stored record" do
       expect {
-        Fabricate(:replication_transfer, stored: true).cancel!('other')
+        Fabricate(:replication_transfer, stored: true).cancel!('other', 'detail')
       }.to raise_error ActiveRecord::RecordInvalid
     end
   end
 
 
+  describe "#cancel!" do
+    let(:transfer) { Fabricate(:replication_transfer) }
+    before(:each) { transfer.cancel!('other', 'test') }
+    it "is idempotent" do
+      expect {
+        transfer.cancel!('other', 'test')
+      }.to_not raise_error
+    end
```

Comment:
This example might add `expect(transfer).to receive(:cancelled).and_call_original` at the top.  If it catches on the count, could add `twice`.